docs/stencil-docs/reference-docs/front-matter-reference.md

```diff
@@ -205,3 +206,151 @@ search:
+  
+ ```handlebars
+ ---
+product:
+    videos:
+        limit: {{theme_settings.productpage_videos_count}}
+    reviews:
+        limit: {{theme_settings.productpage_reviews_count}}
+    related_products:
+        limit: {{theme_settings.productpage_related_products_count}}
+    similar_by_views:
+        limit: {{theme_settings.productpage_similar_by_views_count}}
+ gql: "query productById($productId: Int!) {
+  site {
+    product(entityId: $productId) {
+      variants(first: 25) {
+        edges {
+          node {
+            sku
+            defaultImage {
+              url(width: 1000)
+        }
+       }
+      }
+     }
+    }
+   }
+  }
+  "
+  ```
+We suggest testing GraphQL queries using the [storefront API playground](https://developer.bigcommerce.com/api-reference/storefront/graphql#graphql-playground) to refine them before adding them to your template. You can launch the playground in the context of your store by clicking the **Storefront API Playground** link under the **Advanced Settings** menu in your store's control panel.
+  
+Once you have added a query to your template's front matter block, execution happens automatically when the page loads. The data returned by the query will be returned in the page's context and made available to the handlebars under the `gql` key. For example, you can retrieve the variant data from the above query in `product.html` like this:
+```handlebars
+ {{#if gql.data.site.product}}
+ {{#each gql.data.site.product.variants.edges}}
+    {{#with node}}
+      {{sku}} {{! - - sku code from each variant from GQL response}}
+    {{/with}}
+ {{/each}}
+ {{/if}}
+  ```
+If the GraphQL query is invalid, Stencil returns an `errors` object with `locations` and `message` properties similar to the following example:
+```html
+{
+  "gql": {
+    "errors": [
+      {
+        "locations": [
+          {
+            "column": 1,
+            "line": 1
+          }
+        ],
+        "message": "Syntax error while parsing GraphQL query."
+      }
+    ]
+  }
+}
+```
+  
+On some pages, you can inject special variables into your query to fetch data relevant to that page. For example, using the `$productId` variable on product pages injects the product ID associated with the current page.
+
+The following is the complete list of available variables:
+* `category.html`: `$categoryId`
+* `product.html`: `$productId`
+* `brand.html`: `$brandId`
+* `page.html`: `$pageId`
+* `contact-us.html`: `$pageId`
+* `blog-post.html`: `$blogPostId`
+  
+You can also query data without using variables. The following query returns the product category tree as a JSON object.
+  
+ ```yaml
+gql: "query CategoryTree3LevelsDeep {
+  site {
+    categoryTree {
+      ...CategoryFields
+      children {
+        ...CategoryFields
+        children {
+          ...CategoryFields
+        }
+      }
+    }
+  }
+}
+
+fragment CategoryFields on CategoryTreeItem {
+  name
+  path
+  entityId
+}"
+
+	site {

```

Comment:
did you mean to delete this block?

Reply:
No it should not be deleted. When I view the preview, everything looks good.

Reply:
I accepted your changes to add it.

Reply:
Sorry, I should've suggested a change for the whole block, but the way it is shown right now, what purpose does it serve? To me, it looks like it should be deleted.